tests/kernel/ebpf.pm

```diff
@@ -0,0 +1,101 @@
+# SUSE's openQA tests
+#
+# Copyright 2024 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+#
+# Summary: Executes eBPF kselftests
+# Maintainer: Kernel QE <kernel-qa@suse.de>
+
+use base 'opensusebasetest';
+
+use strict;
+use warnings;
+use testapi;
+use serial_terminal 'select_serial_terminal';
+use registration;
+use utils;
+use LTP::WhiteList;
+use LTP::kirk;
+use LTP::utils;
+
+sub run_tests
+{
+    my ($root) = @_;
+
+    my $env = LTP::utils::prepare_whitelist_environment();
+    my $issues = get_var('KSELFTESTS_KNOWN_ISSUES', '');
+    my $whitelist = LTP::WhiteList->new($issues);
+    my @skipped = $whitelist->list_skipped_tests($env, 'kselftests');
+    my $skip_tests;
+    if (@skipped) {
+        $skip_tests = join("|", @skipped);
+
+        record_info(
+            "Exclude",
+            "Excluding tests: $skip_tests",
+            result => 'softfail'
+        );
+    }
+
+    my @volumes = (
+        {src => $root, dst => $root},
+        {src => "/tmp", dst => "/tmp"}
+    );
+
+    LTP::kirk->run(
+        framework => "kselftests:root=$root",
+        skip => $skip_tests,
+        suite => "bpf",
+        # when KIRK_INSTALL == 'container' we want to share
+        # kselftests folder and kirk logs folder
+        container_volumes => \@volumes,
+    );
+}
+
+sub ebpf_from_source
+{
+    # add repo with newest version of clang
+    zypper_call("ar -f http://download.suse.de/ibs/SUSE:/SLE-15-SP6:/GA/standard/ GA_standard");
+    zypper_call("--gpg-auto-import-keys ref");
+
+    # install dependences
+    zypper_call('install -y bison clang17 llvm17 flex gcc gcc-c++ kernel-devel kernel-default kernel-default-devel kernel-source libcap-devel libopenssl-devel llvm17 make python3-docutils tar rsync bpftool iproute2 iputils');
+
+    # prepare tests compile
+    assert_script_run('cd /usr/src/linux');
+    assert_script_run('cp /boot/config-* .config');
+    assert_script_run('zcat /boot/vmlinux-*.gz > vmlinux');
+    assert_script_run('zstdcat /lib/modules/*/kernel/fs/btrfs/btrfs.ko.zst > fs/btrfs/btrfs.ko');
+
+    # compile kselftests
+    assert_script_run('make tools/bpf/resolve_btfids');
+    assert_script_run("mkdir -p \$HOME/kselftests/bpf");
+    assert_script_run("make install -C tools/testing/selftests/ TARGETS=bpf SKIP_TARGETS= INSTALL_PATH=\$HOME/kselftests/bpf");
+
+    # run tests
+    run_tests("\$HOME/kselftests");
+}
+
+sub ebpf_from_repo
+{
+    my $repo = get_var('BPF_KSELFTESTS_REPO', 'http://download.suse.de/ibs/home:/shunghsiyu:/branches:/SUSE:/SLE-15-SP6:/GA/standard/');
+
+    zypper_call("ar -f $repo ebpf_kselftests");
+    zypper_call("--gpg-auto-import-keys ref");
+
+    zypper_call('install -y kselftests-bpf');
+    run_tests("/usr/share/kselftests");
+}
+
+sub run
+{
+    select_serial_terminal;
+
+    if (get_var("BPF_FROM_SOURCE") eq '1') {
+        ebpf_from_source;
+    } else {
+        ebpf_from_repo;
+    }
```

Comment:
`get_var("BPF_FROM_SOURCE") eq '1' ? ebpf_from_source :  ebpf_from_repo;`  is shorter